src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -32,35 +33,51 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
 
         public override async Task<HttpResponseMessage> ExecuteAsync(HttpControllerContext controllerContext, CancellationToken cancellationToken)
         {
-            HttpRequestMessage request = controllerContext.Request;
-
-            // First see if the request maps to an HTTP function
-            FunctionDescriptor function = _scriptHostManager.GetHttpFunctionOrNull(request);
+            var request = controllerContext.Request;
+            var function = _scriptHostManager.GetHttpFunctionOrNull(request);
             if (function == null)
             {
+                // request does not map to an HTTP function
                 return new HttpResponseMessage(HttpStatusCode.NotFound);
             }
+            request.SetProperty(ScriptConstants.AzureFunctionsHttpFunctionKey, function);
 
-            // Determine the authorization level of the request
-            ISecretManager secretManager = controllerContext.Configuration.DependencyResolver.GetService<ISecretManager>();
-            var settings = controllerContext.Configuration.DependencyResolver.GetService<WebHostSettings>();
-            var authorizationLevel = settings.IsAuthDisabled
-                ? AuthorizationLevel.Admin
-                : await AuthorizationLevelAttribute.GetAuthorizationLevelAsync(request, secretManager, functionName: function.Name);
-            request.SetAuthorizationLevel(authorizationLevel);
-
+            var authorizationLevel = await DetermineAuthorizationLevelAsync(request, function, controllerContext.Configuration.DependencyResolver);
             if (function.Metadata.IsExcluded ||
-                (function.Metadata.IsDisabled && authorizationLevel != AuthorizationLevel.Admin))
+               (function.Metadata.IsDisabled && authorizationLevel != AuthorizationLevel.Admin))
             {
                 // disabled functions are not publicly addressable w/o Admin level auth,
                 // and excluded functions are also ignored here (though the check above will
                 // already exclude them)
                 return new HttpResponseMessage(HttpStatusCode.NotFound);
             }
 
+            Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>> processRequest = async (req, ct) =>
+            {
+                return await ProcessRequestAsync(req, function, ct);
+            };
+            return await _scriptHostManager.HttpRequestManager.ProcessRequestAsync(request, processRequest, cancellationToken);
```

Comment:
This is the main change - the rest is just code refactoring. Here we now dispatch to a request manager which will provide throttling behavior.